consensus-engine/src/types.rs

```diff
@@ -190,6 +190,81 @@ impl Qc {
     }
 }
 
+mod util {
```

Comment:
Maybe we can have a utils module, inside a serde one:
`consensus-engine/utils/serde`?